aten/src/ATen/Parallel.cpp

```diff
@@ -32,10 +67,11 @@ void init_num_threads() {
   }
 }
 
-void set_num_threads(size_t nthreads) {
-  if (nthreads == 0) {
-    return;
+void set_num_threads(int nthreads) {
+  if (nthreads <= 0) {
+    throw std::runtime_error("Expected positive number of threads");
```

Comment:
AT_CHECK looks more suitable here as it is our way to verify use inputs.

Reply:
Always use AT_CHECK and AT_ASSERT etc.